boot/bootutil/src/bootutil_misc.c

```diff
@@ -309,13 +309,13 @@ boot_write_trailer(const struct flash_area *fap, uint32_t off,
     int rc;
 
     align = flash_area_align(fap);
-    if (inlen > BOOT_MAX_ALIGN || align > BOOT_MAX_ALIGN) {
+    if (inlen > BOOT_MAX_ALIGN || align > BOOT_MAX_ALIGN ||
+	BOOT_MAX_ALIGN % align) {
```

Comment:
I'd be more comfortable with: `.. || BOOT_MAX_ALIGN % align != 0` just to make it clearer. But this is ok as well.

Reply:
Probably more efficient in runtime using `BOOT_MAX_ALIGN & (align - 1)`

Reply:
Will change

Reply:
Sorry about this, but it was a wrong suggestion on my part; that does not work, the correct way to do the same as the modulo that you done originally is:

```
... || (align <= BOOT_MAX_ALIGN && (align & (align - 1)) == 0) {
```

I am not sure that is really something that needs protection against though...

Reply:
Not my day as well I guess, just need to add this extra || test `(align & (align - 1)) != 0`...